File: test/standalone/java/PhoneCalls.java
package generated;

import grakn.client.GraknClient;
import graql.lang.Graql;
import graql.lang.query.GraqlQuery;
import grakn.core.rule.GraknTestServer;

import java.io.FileNotFoundException;
import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Paths;

import org.junit.*;
import org.junit.runners.MethodSorters;

import javax.xml.stream.XMLStreamException;

@FixMethodOrder(MethodSorters.NAME_ASCENDING)
public class TestStandalonePhoneCalls {

    @ClassRule
    public static final GraknTestServer server = new GraknTestServer(
            Paths.get("test/grakn-test-server/conf/grakn.properties"),
            Paths.get("test/grakn-test-server/conf/cassandra-embedded.yaml")
    );

    @BeforeClass
    public static void loadPhoneCalls() {
        GraknClient client = new GraknClient(server.grpcUri().toString());
        GraknClient.Session session = client.session("phone_calls");
        GraknClient.Transaction transaction = session.transaction().write();

        try {
            byte[] encoded = Files.readAllBytes(Paths.get("files/phone-calls/schema.gql"));
            String query = new String(encoded, StandardCharsets.UTF_8);
            transaction.execute((GraqlQuery) Graql.parse(query));
            transaction.commit();
            session.close();
            client.close();
            System.out.println("Loaded the phone_calls schema");
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    @Test
    public void testAPhoneCallsFirtstQuery() {
        String[] args = { server.grpcUri().toString() };
        PhoneCallsFirstQuery.main(args);
    }

    @Test
    public void testBPhoneCallsSecondQuery() {
        String[] args = { server.grpcUri().toString() };
        PhoneCallsSecondQuery.main(args);
    }

    @Test
    public void testCPhoneCallsThirdQuery() {
        String[] args = { server.grpcUri().toString() };
        PhoneCallsSecondQuery.main(args);
    }

    @Test
    public void testDPhoneCallsForthQuery() {
        String[] args = { server.grpcUri().toString() };
        PhoneCallsForthQuery.main(args);
    }

    @Test
    public void testEPhoneCallsFifthQuery() {
        String[] args = { server.grpcUri().toString() };
        PhoneCallsFifthQuery.main(args);
    }

    @Test
    public void testFPhoneCallsCSVMigration() throws FileNotFoundException {
        String[] args = { server.grpcUri().toString() };
        PhoneCallsCSVMigration.main(args);
    }

    @Test
    public void testGPhoneCallsJSONMigration() throws IOException {
        String[] args = { server.grpcUri().toString() };
        PhoneCallsJSONMigration.main(args);
    }

    @Test
    public void testHPhoneCallsXMLMigration() throws FileNotFoundException, XMLStreamException {
        String[] args = { server.grpcUri().toString() };
        PhoneCallsXMLMigration.main(args);
    }

    @AfterClass
    public static void cleanPhoneCalls() {
        GraknClient client = new GraknClient(server.grpcUri().toString());
        client.keyspaces().delete("phone_calls");
        client.close();
        System.out.println("Deleted the phone_calls keyspace");
    }
}

File: test/standalone/python/phone_calls.py
import grakn
import unittest

class TestStandalonePhoneCalls(unittest.TestCase):

    @classmethod
    def setUpClass(cls):
        with open('files/phone-calls/schema.gql', 'r') as schema:
            define_query = schema.read()

            client = grakn.Grakn(uri="localhost:48555")
            with client.session(keyspace="phone_calls") as session:
                with session.transaction(grakn.TxType.WRITE) as transaction:
                    transaction.query(define_query)
                    transaction.commit()

    def test_a_phone_calls_first_query(self):
        import phone_calls_first_query

    def test_b_phone_calls_second_query(self):
        import phone_calls_second_query

    def test_c_phone_calls_third_query(self):
        import phone_calls_third_query

    def test_d_phone_calls_forth_query(self):
        import phone_calls_forth_query

    def test_e_phone_calls_fifth_query(self):
        import phone_calls_fifth_query

    def test_f_phone_calls_csv_migration(self):
        import phone_calls_csv_migration

    def test_g_phone_calls_json_migration(self):
        import phone_calls_json_migration

    def test_h_phone_calls_xml_migration(self):
        import phone_calls_xml_migration

    @classmethod
    def tearDown(cls):
        client = grakn.Grakn(uri="localhost:48555")
        with client.session(keyspace="phone_calls") as session:
            with session.transaction(grakn.TxType.WRITE) as transaction:
                transaction.query("match $x isa relationship; delete $x;")
                transaction.query("match $x isa entity; delete $x;")
                transaction.query("match $x isa attribute; delete $x;")
                transaction.commit()

if __name__ == '__main__':
    unittest.main()
